### PR TITLE
simple changes to mxnet.image.ImageIter to make it more newbie friendly

### PR DESCRIPTION
**Pull request by jmerkow**
This pull request makes it a bit easier to build off of mx.io.ImageIter to make your own image iters by pulling out some code out into their own functions.

Example:

```
class RandImageIter(mx.image.ImageIter):
    
    def check_data_shape(self, data_shape):
        if not len(data_shape) == 3:
            raise ValueError('data_shape should have length 3, with dimensions CxHxW')
        if not data_shape[0] == 1:
            raise ValueError('This iterator expects inputs to have 3 channels.')
    
    def imdecode(self, s):
        return mx.nd.array(s)
    
    def read_image(self, fname):
        #fname is disregarded for brevity in example    
        return np.random.random(self.data_shape[::-1])
```

### CODE REVIEW

**Comment by jmerkow** [comment]
The CI failed due to a docker error:

> Unable to find image 'mxnet/aml:latest' locally
> Pulling repository docker.io/mxnet/aml
> docker: Error: image mxnet/aml:latest not found.

File: python/mxnet/image.py [hunk 1]
```diff
@@ -437,19 +441,53 @@ def next(self):
...
+            img = fin.read()
+        return image
+    
+    def prepare_augmentation(self, data):
+        """Prepares augmentation of data."""
+        for aug in self.auglist:
+            data = [ret for src in data for ret in aug(src)]
+        return data
+            
+    def preprocess_data(self, datum):
```
Comment (piiswrong) [inline python/mxnet/image.py]: sounds more like postprocessing

File: python/mxnet/image.py [hunk 2]
```diff
@@ -437,19 +441,53 @@ def next(self):
...
+        """Decodes a sting or byte string into an image."""
+        return imdecode(s)
+    
+    def read_image(self, fname):
+        """Reads image from fname and returns the raw bytes to be decoded."""
+        with open(os.path.join(self.path_root, fname), 'rb') as fin:
+            img = fin.read()
+        return image
+    
+    def prepare_augmentation(self, data):
```
Comment (piiswrong) [inline python/mxnet/image.py]: its actually doing the augumentation, not preparing for it. How about transform.

**Comment by piiswrong** [comment]
Can one of the admins verify this patch?

**Comment by piiswrong** [comment]
please fix lint

**Comment by jmerkow** [comment]
*should* be good now.

**Comment by jmerkow** [comment]
The lint errors look to me issues with the merge you added?  I didn't not add/change any imports...

**Comment by jmerkow** [comment]
I moved that import,  I can revert if it was placed at the top for a reason.

**Comment by jmerkow** [comment]
@piiswrong I think this is ready to merge now.

**Pull request merged**